cpp/src/components/connectivity.cu

```diff
@@ -41,190 +40,71 @@ namespace detail {
-      cudaMemcpy((void*)(&last_elem), p_d_row_offsets+nrows, sizeof(IndexT), cudaMemcpyDeviceToHost);
-      std::cout<<"############## "
-               <<"nrows = "<<nrows
-               <<"; nnz = "<<nnz
-               <<"; nnz_ro = "<<last_elem
-               <<"; p_d_labels valid: "<<(p_d_labels != nullptr)
-               <<"; p_d_row_offsets valid: "<<(p_d_row_offsets != nullptr)
-               <<"; p_d_col_ind valid: " << (p_d_col_ind != nullptr)
-               <<"\n";
-      
-      std::cout<<"############## d_ro:\n";
-      print_v(d_ro, std::cout);
-
-      std::cout<<"############## d_ci:\n";
-      print_v(d_ci, std::cout);
-#endif
-
+  if (connectivity_type == cugraph_cc_t::CUGRAPH_WEAK) {
       //check if graph is undirected; return w/ error, if not?
       //Yes, for now; in the future we may remove this constraint; 
       //
-      bool is_symmetric = cugraph::detail::check_symmetry(nrows, p_d_row_offsets, nnz, p_d_col_ind);
-#ifdef _DEBUG_WEAK_CC
-      std::cout<<"############## "
-               <<"; adj. matrix symmetric? " << is_symmetric
-               <<"\n";
-#endif
+      bool is_symmetric = cugraph::detail::check_symmetry(graph.number_of_vertices, graph.offsets, graph.number_of_edges, graph.indices);
+
+      auto d_alloc = std::shared_ptr<MLCommon::deviceAllocator>{new MLCommon::defaultDeviceAllocator()};
       
       CUGRAPH_EXPECTS( is_symmetric, "Invalid API parameter: graph must be symmetric");
-      MLCommon::Sparse::weak_cc_entry<IndexT, TPB_X>(p_d_labels,
-                                                     p_d_row_offsets,
-                                                     p_d_col_ind,
-                                                     nnz,
-                                                     nrows,
+      MLCommon::Sparse::weak_cc_entry<VT, ET, TPB_X>(labels,
+                                                     graph.offsets,
+                                                     graph.indices,
+                                                     graph.number_of_edges,
+                                                     graph.number_of_vertices,
+                                                     d_alloc,
                                                      stream);
-
-    }
-  else
-    {
-      
-      //device memory requirements: 2n^2 + 2n x sizeof(IndexT) + 1 (for flag)
-      //( n = |V|)
-      //
-      size_t n2 = 2*nrows;
-      n2 = n2*(nrows*sizeof(ByteT) + sizeof(IndexT)) + 1;
-
-      int device;
-      cudaDeviceProp prop;
+  } else {
+    //device memory requirements: 2n^2 + 2n x sizeof(IndexT) + 1 (for flag)
+    //( n = |V|)
+    //
+    size_t n2 = 2 * graph.number_of_vertices;
+    n2 = n2 * (graph.number_of_vertices * sizeof(ByteT) + sizeof(VT)) + 1;
+
+    int device;
+    cudaDeviceProp prop;
       
-      cudaGetDevice(&device);
-      cudaGetDeviceProperties(&prop, device);
+    cudaGetDevice(&device);
+    cudaGetDeviceProperties(&prop, device);
```

Comment:
We should avoid cudaGetDeviceProperties to get just a single attribute.

https://devblogs.nvidia.com/cuda-pro-tip-the-fast-way-to-query-device-properties/

"cudaDeviceGetAttribute() is a much faster way to query device properties."